lib/auth/init.go

```diff
@@ -1405,6 +1408,15 @@ func applyResources(ctx context.Context, service *Services, resources []types.Re
 			err = service.ClusterConfiguration.SetClusterNetworkingConfig(ctx, r)
 		case types.AuthPreference:
 			err = service.ClusterConfiguration.SetAuthPreference(ctx, r)
+		case interface{ Unwrap() types.Resource153 }:
```

Comment:
FYI @codingllama - this may be of interest to you in terms of how the Adapter works in practice.

Reply:
Thanks for copying me. If this is the only Unwrap then I suppose I'm somewhat OK with it, but let me know if you need to add more.

It also looks like a switch over kind is possible. The semantics might be better, but the type assertions are still necessary.

This could also be candidate for migration to a smaller interface, as it doesn't really seem to need anything apart from Kind. FYI @rosstimothy.

Reply:
Thanks Alan.

> This could also be candidate for migration to a smaller interface, as it doesn't really seem to need anything apart from Kind

I do agree - this is why one of my suggestions is that instead of having `types.ResourceMetadata`, we actually probably need something like `types.ResourceVersionKind`.

Reply:
I'm perfectly OK with smaller, impromptu interfaces for cases like this. I don't think we should export all permutations of resource-like interfaces, like the various ResourceWithFoo we have now. ResourceMetadata is something you can use as a proxy for "this is a resource", while making few assumptions about it.